src/app/games/encapsule/EncapsuleRules.ts

```diff
@@ -53,35 +52,42 @@ export class EncapsuleRules extends Rules<EncapsuleMove, EncapsuleState, Encapsu
         [new Coord(0, 2), new Coord(1, 1), new Coord(2, 0)],
     ];
 
-    public static isVictory(state: EncapsuleState): MGPOptional<Player> {
-        const board: EncapsuleSpace[][] = state.getCopiedBoard();
+    public getVictoriousCoords(state: EncapsuleState): Coord[] {
         for (const line of EncapsuleRules.LINES) {
-            const spaces: EncapsuleSpace[] = [board[line[0].y][line[0].x],
-                board[line[1].y][line[1].x],
-                board[line[2].y][line[2].x]];
-            const victory: MGPOptional<Player> = EncapsuleRules.isVictoriousLine(spaces);
-            if (victory.isPresent()) {
-                return victory;
+            const spaces: EncapsuleSpace[] = [
+                state.getPieceAt(line[0]),
+                state.getPieceAt(line[1]),
+                state.getPieceAt(line[2]),
+            ];
+            const victory: boolean = this.isVictoriousLine(spaces);
```

Comment:
C'pas une ligne ça ! passe la ligne ou appelle ça "are all the same player", si possible le premier :D